### PR TITLE
[Tooling] [Screenshots] Update documentation for screenshots

### PR DESCRIPTION
**Pull request by AliSoftware**
> **Note**: This PR builds on top of https://github.com/wordpress-mobile/WordPress-Android/pull/17505 — since we want the new documentation to live under `fastlane/screenshots/README.md` but the new and un-ignored `fastlane/screenshots` folder will only exist after #17505 lands

---

This PR:
 - Removes obsolete documentation about how the `SCREENSHOT_DEVICE_SETUP.md` worked, now that we've modernized the way we create emulators on the fly using ruby code, instead of the old way of doing it using `.sh` script and `.ini` template files (see changes made in #17096)
 - Adds new documentation to explain the various steps to update and generate the screenshots and how the generation automated process works.

### CODE REVIEW

File: fastlane/screenshots/README.md
Comment (mokagio) [inline fastlane/screenshots/README.md]:
```suggestion
The screenshots used for the Play Store come in multiple device sizes (phone vs tablet) and many locales (currently 16 different languages), so it would be unreasonable to expect designers to generate all the variants by hand every time we want to update those screenshots. For this reason, the generation of those Play Store screenshots is automated with Fastlane.
```

File: fastlane/screenshots/README.md
Comment (mokagio) [inline fastlane/screenshots/README.md]: How does this look like? Just a sentence structure tweak, although I think the existing version is still correct.

```suggestion
 - First, we have UI tests in the codebase — namely [`WPScreenshotTest`](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/WPScreenshotTest.java) and [`JPScreenshotTest`](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/JPScreenshotTest.java) — which are responsible for launching the app, navigating through the various screens we need screenshot for, and taking a "raw screenshot" (screen capture) of those screens.
```

File: fastlane/screenshots/README.md
Comment (mokagio) [inline fastlane/screenshots/README.md]: I suggest removing "as the app evolved" to get to the point of the item (update the tests) faster.

```suggestion
    - If you plan to update the Play Store screenshots to include different screens that the ones currently included, or if the navigation of the app to get to those screens changed, you'll need to update those `WPScreenshotTest` and `JPScreenshotTest` test suites accordingly.
```

File: fastlane/screenshots/README.md
Comment (mokagio) [inline fastlane/screenshots/README.md]:
```suggestion
 - Then, the lane `screenshots` is responsible for:
```

For consistency with a few lines below, were it uses the form "lane `create_promo_screenshots`", without `fastlane`. IMHO, that's better because it's shorter and "correct" from the point of view that `screenshot` is the lane name, while `[bundle exec] fastlane screenshot` is how we'd invoke it.

File: fastlane/screenshots/README.md
Comment (mokagio) [inline fastlane/screenshots/README.md]:
```suggestion
 - Once we have those "raw screenshots" captured from the app screens in all the locales, the lane `create_promo_screenshots` assembles (doing image composition, using `imagemagick`) those raw screenshots in a background image and adding a device frame + some (localized) annotation text from the `screenshot_{N}.txt` files. The way to assemble those elements (paths to the files, coordinates where to place those in final image, etc.) is described in the `fastlane/screenshots/{app}-config.json` file.
```

File: fastlane/screenshots/README.md
Comment (mokagio) [inline fastlane/screenshots/README.md]:
```suggestion
> This also sadly comes with risks of having installation troubles depending on the environment of your machine (and which libraries and headers you have installed, etc); in fact, it's common to have issues when trying to `brew install imagemagick` or to install `rmagick`, especially on the steps which tries to compile those libraries (since they rely on compiled, binary code and libraries) and might come up with compilation errors or missing headers and the like. There's sadly no magic silver bullet solution for those, and search engines and StackOverflow are usually your best friends here.
```

- Nitpick: "sadly" already used at the start of the sentence
- Nitpick: removed double negation
- Nitpick: removed potential controversy on whether to pick Google or DDG

**Review by mokagio — COMMENTED** [review]
Only left a few nitpicks.

**Review by mokagio — APPROVED** [review]
I actually meant to approve, not [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/17395#pullrequestreview-1195814555) on this...

**Comment by AliSoftware** [comment]
Thanks @mokagio for the review! All great re-wording suggestions 👍 

Note: Since this PR targets the head branch of #17505 and stacks on top of it, I will wait for #17505 to land first before merging this one though, to avoid fusing the diffs and to keep the review of #17505 easier.

**Pull request merged**
